src/coreclr/nativeaot/Runtime/gcrhscan.cpp

```diff
@@ -53,6 +54,9 @@ void GCToEEInterface::GcScanRoots(EnumGcRefCallbackFunc * fn,  int condemned, in
         else
 #endif
         {
+            STRESS_LOG1(LF_GC | LF_GCROOTS, LL_INFO100, "{ Scanning Thread's %p thread statics root. \n", pThread);
+            GcEnumObject(pThread->GetThreadStaticStorage(), 0 /*flags*/, fn, sc);
```

Comment:
Here we report the location of threadstatic root to the GC - to keep the array alive and the pointer updated as long as the owner thread is alive. 
Just another root reported by the thread.